### DISCUSSION TITLE
Memory increased to 50%

### DISCUSSION THREAD

**Discussion by philzhangrui** [apache/apisix#5447] [Q&A]
In the configuration file, the following parameters are used
proxy_buffering on;
proxy_buffer_size 4k;
proxy_buffers 8 512k;
proxy_busy_buffers_size 1M;
proxy_max_temp_file_size 0;
client_body_buffer_size 30M;

A few days later，memory increased to 50%

**Comment by tzssangglass** [comment]
maybe related to:https://github.com/apache/apisix/pull/4405

**Comment by tokers** [comment]
> In the configuration file, the following parameters are used proxy_buffering on; proxy_buffer_size 4k; proxy_buffers 8 512k; proxy_busy_buffers_size 1M; proxy_max_temp_file_size 0; client_body_buffer_size 30M;
> 
> A few days later，memory increased to 50%

What's the version of your Apache APISIX?

**Reply by philzhangrui** [reply]
version 2.10.0

**Reply by tokers** [reply]
We have to know several data before we can troubleshoot this problem, for instance, what's the CPU utilization, what's the QPS .

**Reply by philzhangrui** [reply]
CPU utilization is relatively low and there are many uploaded files

**Reply by tzssangglass** [reply]
If APISIX is at 50% memory usage even when it is idle, I suspect a memory leak is occurring. Can you provide a stable reproduction step?

**Reply by philzhangrui** [reply]
I'll try it tomorrow. It's coming back
